app/views/schools/calendar_events/_form.html.erb

```diff
@@ -39,7 +42,7 @@
         <div class="col-span-6">
           <label>
             <span class="text-sm font-medium"><%=t(".calendar_label")%></span>
-            <%= f.select :calendar_id, options_from_collection_for_select(course.calendars, 'id', 'name'), { include_blank: false }, { class: "select mt-1 block w-full rounded-md bg-white border border-gray-300 py-3 px-4 pe-10 text-base focus:outline-none focus:ring-1 focus:ring-focusColor-500 sm:text-sm" } %>
+            <%= f.select :calendar_id, options_from_collection_for_select(course.calendars, 'id', 'name', selected: (params[:calendar_id] || @event.calendar_id)), { include_blank: false }, { class: "select mt-1 block w-full rounded-md bg-white border border-gray-300 py-3 px-4 pe-10 text-base focus:outline-none focus:ring-1 focus:ring-focusColor-500 sm:text-sm" } %>
```

Comment:
I think there is an issue here. If it's an edit operation, i.e if `@event.persisted?` we should always have the selected value as `@event.calendar_id`. Otherwise, you create a possibility of having a different value in the `calendar_id` param and that gets selected if present.